### PR TITLE
docs: fix example of `.blocks.det` to `.hap` conversion in API docs

### LINKED ISSUE #235: Provided blocks.det to .hap code produces error
Hello 

The code provided on your website for converting the plink output (*blocks.det) into the haps format does not parse the blocks.det file properly.

I encountered several errors in trying to run an example dataset. Can you please confirm that the code on the webpage is the current working version?

I am referring to this code: https://haptools.readthedocs.io/en/stable/api/examples.html

Thanks!

### PR DESCRIPTION
**Pull request by aryarm**
resolves #235 

There were a few bugs in the blocks2hap example in the API docs!

In d06d50a27b1ae847a044e9bb287ed186c58aa53f, we fix those bugs. And in aa5d4efebe9bce35846a75a74951c63fc358452d, we convert that example code into a test so we can properly maintain it going forward.

### CODE REVIEW

**Comment by aryarm** [comment]
The tests successfully pass on my local machine, so I think this is just a CI issue

The CI is revamped in #207 , so merging that will probably resolve these issues, anyway. Let's just wait until that's merged before merging this

**Comment by mlamkin7** [comment]
> The tests successfully pass on my local machine, so I think the CI is just having another hissy fit
> 
> The CI is revamped in #207 , so merging that will probably resolve these issues, anyway. Let's just wait until that's merged before trying to merge this

Sounds good to me, I'll approve anyway so you can merge when ready.

**Review by mlamkin7 — APPROVED** [review]

**Pull request merged**
